Feature/party add view

ti-tanino opened the pull request:
飲み会一覧画面
・カスタムセルで開催日を表示
・削除ボタンを追加、削除機能

Commit:
feature確認

Commit:
飲み会作成画面　閉じるボタン配置

Commit:
飲み会TBL作成　TBL読み込み時点で頓挫

Commit:
【修正点】
・飲み会オブジェクトの名称からTBLを削除
・オブジェクトの宣言形式を修正し、初期値を与える
・プロジェクトにParty.swiftを認識させる
・テスト呼び出しのオブジェクト宣言位置を変更

Commit:
テストコードを削除
飲み会作成画面の閉じるボタンと
Partyオブジェクトをマージしたいです。

ti-tanino committed:
飲み会作成画面　部品配置

ti-tanino committed:
入力項目、ボタンとソースコードの紐付け

ti-tanino committed:
飲み会追加画面からのデータ追加

ti-tanino committed:
飲み会一覧画面でデータ件数分の表示及びデータ取得

ti-tanino committed:
Merge branch 'develop' into feature/PartyAddView

ti-tanino committed:
マージ及び競合解消

ti-tanino committed:
Merge branch 'feature/PartyAddView' of ssh://github.com/TechLabGroup/duesCalculateApp into feature/PartyAddView

ti-tanino committed:
ワーニングの修正

ti-tanino committed:
飲み会データ追加後に一覧画面に反映されるようにテーブルの更新処理を追加

ti-tanino committed:
飲み会一覧　カスタムセルで日付表示追加
　　　　　　削除機能を追加

nagasakiTechLab commented on DuesCalculateApp/Model/DB/DBManager.swift:
```diff
@@ -10,4 +10,39 @@ import RealmSwift
+    public func searchParty() -> Results<Party> {
+        let realm = try! Realm()
+        let party = realm.objects(Party.self)
+        return party
```
Realmの世界観は/Modle/DBの中だけに留めておくべきなので、 `return Array(party)` で単なる配列にして返してあげる方がよい

nagasakiTechLab commented on DuesCalculateApp/View/PartyAdd/PartyAddViewController.swift:
```diff
@@ -9,29 +9,54 @@ import UIKit
+        let partyDate = inputPartyDate.text
+        
+        
+        let totalAmount: Int? = Int(inputTotalAmount.text!)
```
`: Int?` は不要なはず

nagasakiTechLab commented on DuesCalculateApp/Model/DB/DBManager.swift:
```diff
@@ -10,4 +10,39 @@ import RealmSwift
+        return party
+    }
+    
+    public func deleteParty(party: Results<Party>, indexId: Int) {
```
ここも同じく、引数にResults型を持たない。
再検索してから削除かな。

nagasakiTechLab commented on DuesCalculateApp/View/PartyAdd/PartyAddViewController.swift:
```diff
@@ -9,29 +9,54 @@ import UIKit
     }
     
+    // MARK: - private method
+    func tapCloseButton() {
```
公開範囲はprivateでいいと思う。ただ `#selector` の対象にしているので、 `@objc` のキーワードもつける必要があります。まとめると以下です。
```
@objc private func tapCloseButton() {
```

nagasakiTechLab replied:
PartyListViewControllerの横並び確認もお願いします

nagasakiTechLab commented on DuesCalculateApp/View/PartyAdd/PartyAddViewController.swift:
```diff
@@ -9,29 +9,54 @@ import UIKit
+        
+        let totalAmount: Int? = Int(inputTotalAmount.text!)
+        
+        let _: Bool = DBManager().createParty(partyName: partyName!, partyDate: partyDate!, totalAmount: totalAmount!)
```
値精査をしたら `!` によるforceUnwrappingは不要になるはずですが、忘れないようにコメントで `// TODO: 精査後要修正` を書いておいてください。

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyListViewController.swift:
```diff
@@ -74,17 +99,19 @@ class PartyListViewController: UIViewController, UITableViewDelegate, UITableVie
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            // Delete the row from the data source
+            DBManager().deleteParty(party: object, indexId: indexPath.row)
```
indexPath.rowでレコード特定するなら少なくともsearchPartyおよびdeletePartyで、レコード検索結果のsort指定はするべきですね。

nagasakiTechLab replied:
ほんとはキーであるIDとかで指定するべき。検討してみてください。

nagasakiTechLab reviewed (changes requested):
Swift上達してるようで何より。
コメント参照で対応お願いします。

ti-tanino committed:
プルリク指摘修正

ti-tanino committed:
プルリクの指摘対応完了

ti-tanino commented:
@nagasakiTechLab 
指摘対応しました！再確認お願いしますー

nagasakiTechLab commented on DuesCalculateApp/Model/DB/DBManager.swift:
```diff
@@ -31,16 +31,17 @@ final class DBManager {
+        let party = realm.objects(Party.self)
         try! realm.write {
-            realm.delete(party[indexId])
+            realm.delete(party[partyId])
```
party[]の中に入る数字って単なる順番ですよね？
Idと順番はイコールではないですね。
42行目で検索結果をpartyIdでfilterしてください。

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyListViewController.swift:
```diff
@@ -24,7 +24,7 @@ class PartyListViewController: UIViewController, UITableViewDelegate, UITableVie
         self.navigationItem.rightBarButtonItem = rigthItem
         
-        object = DBManager().searchParty()
+        parties = DBManager().searchParty()
```
DBManagerインスタンス化しまくってたんですね。
DBManagerのファンクションはstaticにしましょう。

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyListViewController.swift:
```diff
@@ -10,7 +10,7 @@ import RealmSwift
 class PartyListViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
     
-    var object: Results<Party>!
+    var parties: [Party]!
```
`!` は基本的に使わないようにしましょう。
めんどくさくてもオプショナル（ `?` ）にしてください。

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyTableViewCell.swift:
```diff
@@ -11,6 +11,7 @@ import UIKit
 class PartyTableViewCell: UITableViewCell {
     @IBOutlet weak var date: UILabel!
     @IBOutlet weak var partyName: UILabel!
+    var partyId: Int = 0
```
ここもめんどくさくてもオプショナルにするべき。初期値0に意味がない。

nagasakiTechLab reviewed (changes requested):
確認しました。まだ修正必要です。

ti-tanino committed:
DBManager  関数のstatic化
deletePartyの主キーによる削除が不正だったため単一オブジェクトを検索して削除に修正
選択したcellの値が正しく取得できていなかったため取得方法を修正及び削除後の保有オブジェクトを更新する処理追加
変数宣言時の暗黙的アンラップ型の排除

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyListViewController.swift:
```diff
@@ -63,26 +63,19 @@ class PartyListViewController: UIViewController, UITableViewDelegate, UITableVie
         
         // 通常は引数のセクションで分岐して値を返却する
-        return parties.count
+        return parties!.count
```
`!` は避けましょう。↓こんな感じで万が一nullだったときの値を設定してあげればよい（詳しくは本日のSwift勉強会で高崎くんが教えてくれるはずｗ）
```
return parties?.count ?? 0
```

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyListViewController.swift:
```diff
@@ -63,26 +63,19 @@ class PartyListViewController: UIViewController, UITableViewDelegate, UITableVie
-     // Configure the cell...
+        cell.partyName?.text = parties?[indexPath.row].partyName
+        cell.date?.text = parties?[indexPath.row].date
+        cell.partyId = (parties?[indexPath.row].partyId)!
```
partyIdをオプショナルで宣言してるので `!` しなくてもいいはず

nagasakiTechLab commented on DuesCalculateApp/View/PartyList/PartyTableViewCell.swift:
```diff
@@ -9,9 +9,9 @@
-    @IBOutlet weak var partyName: UILabel!
-    var partyId: Int = 0
+    @IBOutlet weak var date: UILabel?
+    @IBOutlet weak var partyName: UILabel?
```
@IBOutletは元から `!` で宣言されてると思うのでそのままで大丈夫です

nagasakiTechLab commented on DuesCalculateApp/Model/DB/DBManager.swift:
```diff
@@ -11,7 +11,7 @@ import RealmSwift
     
     // 飲み会作成
-    public func createParty(partyName: String, partyDate: String, totalAmount: Int) -> Bool {
+    static public func createParty(partyName: String, partyDate: String, totalAmount: Int) -> Bool {
```
一般的には public static の順番ですね（public static void main的に考えて）

nagasakiTechLab reviewed (changes requested):
コメント参照ください

ti-tanino committed:
指摘対応実施

nagasakiTechLab commented:
OKです！
